tests/components/pegel_online/test_config_flow.py

```diff
@@ -0,0 +1,165 @@
+
+        await hass.async_block_till_done()
+
+    assert mock_setup_entry.called
+
+
+async def test_user_already_configured(hass: HomeAssistant) -> None:
+    """Test starting a flow by user with an already configured statioon."""
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_CONFIG_ENTRY_DATA,
+        unique_id=MOCK_CONFIG_ENTRY_DATA[CONF_STATION],
+    )
+    mock_config.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.pegel_online.config_flow.PegelOnline",
+    ) as pegelonline:
+        pegelonline.return_value = PegelOnlineMock(nearby_stations=MOCK_NEARBY_STATIONS)
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA_STEP1
+        )
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "select_station"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA_STEP2
+        )
+        assert result["type"] == FlowResultType.ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_connection_error(hass: HomeAssistant) -> None:
+    """Test connection error during user flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.pegel_online.config_flow.PegelOnline",
+    ) as pegelonline:
+        pegelonline.return_value = PegelOnlineMock(side_effect=ClientError)
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA_STEP1
+        )
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "cannot_connect"
+
+
+async def test_user_no_stations(hass: HomeAssistant) -> None:
+    """Test starting a flow by user which does not find any station."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.pegel_online.config_flow.PegelOnline",
+    ) as pegelonline:
+        pegelonline.return_value = PegelOnlineMock(nearby_stations={})
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA_STEP1
+        )
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"][CONF_RADIUS] == "no_stations"
```

Comment:
A user could try again / errors can be recovered. Continue the tests until the config entry is created to ensure the errors have no side-effects.